### PR TITLE
Use generated certs.jar instead of enterprise-tests.jar [API-1766]

### PR DESCRIPTION
**Pull request by mdumandag**
We no longer generate the enterprise-tests.jar, starting from 5.2.0.

As a solution, we have decided to generate the certificates used in tests in a separate repo, and use the certificates from there.

We can easily do that for the new tests, but for the tests here, we have to continue pretending as if there is an enterprise-tests jar because we cannot change the run scripts of the already released clients.

Luckily, all of the scripts are smart enough to not try to download a jar, if it exists locally. We can make use of this and pretend like the certs.jar is the enterprise-tests.jar. We have produced certs.jar to be a drop-in replacement for the enterprise-tests.jar and it has the certificates used by clients in the same path as the enterprise-tests.jar.

In this PR, we download the cert.jar from the private-test-artifacts repo, rename it as necessary, and start the remote controller with it, hoping that the run scripts of the clients will no longer try to download the existing jar.

### CODE REVIEW

File: .github/workflows/server_compatibility.yaml [hunk 1]
```diff
@@ -96,11 +99,20 @@ jobs:
...
           retention-days: 1
-      - name: Upload Hazelcast Enterprise tests JAR
+      - name: Rename certs.jar as Hazelcast Enterprise tests JAR
+        # - Find a file named "hazelcast*-tests.jar" in the $GITHUB_WORKSPACE/hazelcast/hazelcast/target -> find command
+        # - Get rid of the file path, so that we only have the file name -> piped sed command
+        # - Replace the hazelcast with hazelcast-enterprise so that we can end up with the expected
+        #   enterprise test file name -> parameter expansion
+        # - Rename the certs.jar with that -> mv command
+        run: |
+          mv certs.jar "${$(find $GITHUB_WORKSPACE/hazelcast/hazelcast/target -maxdepth 1 -mindepth 1 -name "hazelcast*" -a -name "*tests.jar" | sed "s/.*\///")//hazelcast/hazelcast-enterprise}"
```
Comment (srknzl) [inline .github/workflows/server_compatibility.yaml:109]: Did not get this command fully.

what about this? 

```suggestion
          mv certs.jar $(basename $(ls hazelcast/target/hazelcast-*-tests.jar))
```

basename exists by default in ubuntu.

Comment (srknzl) [reply .github/workflows/server_compatibility.yaml:109]: Anyway, I am ok if it works. Can we try to run all backward compatilbity tests once? (server and all clients)

**Review by srknzl — COMMENTED** [review]
Looks good in general, see my review

File: .github/workflows/server_compatibility.yaml (hunk 1 above)
Comment (mdumandag) [reply .github/workflows/server_compatibility.yaml:109]: Let's try this as it is, and if it does not work we can try yours. The comment above should explain what each part does

**Review by srknzl — APPROVED** [review]

**Pull request merged**